### PR TITLE
Data.BaseXX improvement.

### PR DESCRIPTION
**Pull request by tsuyoshicho**
This PR is improve below:

- Data.Base16 imported vital-codec lib. (based routine is Data.List.Byte)
- Data.Base32 and variant imported  vital-codec lib. (refactoring:aggregate internal routine)
  - side effect : Base32 RFC test code improved
- Data.Base64 and variant imported vital-codec lib. (refactoring:aggregate internal routine)
  - side effect : Base64 RFC test code improved

NOTE:if need PR split. I create per commit PRs.

### CODE REVIEW

**Comment by ujihisa** [comment]
(I found that @tsuyoshicho was not github vital.vim team, so I added him/her.)

**Comment by ujihisa** [comment]
> Data.Base16 imported vital-codec lib. (based routine is Data.List.Byte)

hmm, I got confused. Looks like Data.Base16 did not exist in vital.vim right?

File: README.md [hunk 1]
```diff
@@ -42,8 +42,14 @@ Module						 | Description
 [Async.Promise](doc/vital/Async/Promise.txt)	 | An asynchronous operation like ES6 Promise
 [Bitwise](doc/vital/Bitwise.txt)		 | bitwise operators
 [ConcurrentProcess](doc/vital/ConcurrentProcess.txt)	 | manages processes concurrently with vimproc
-[Data.Base64](doc/vital/Data/Base64.txt)	 | base64 utilities library
-[Data.Base32](doc/vital/Data/Base32.txt)	 | base32 utilities library
+[Data.Base64](doc/vital/Data/Base64.txt)	 | base64 utilities library (RFC Wrapper IF)
```
Comment (ujihisa) [inline README.md]: What does RFC wrapper wraps Base64 for?

Comment (tsuyoshicho) [reply README.md]: I think, need old IF and save it.
If do not so, remove it Okay.

**Comment by tsuyoshicho** [comment]
> hmm, I got confused. Looks like Data.Base16 did not exist in vital.vim right?

In addition, I created it because I thought it would be better to have Base16 with the same IF as Base32 and 64.

* Base16 is not required (Data.List.Byte can be used directly)
* Base16 is included separately PR
* Base16 is included this OK

I think either is fine.

File: README.md (hunk 1 above)
Comment (tyru) [reply README.md]: If the interface is for backward compatibility, it is not "RFC Wrapper"?
Did you mean Base64 module satisfies RFC of base64?
If so, "RFC Wrapper IF" is not necessary.
It is enough to be written in help, I think.

Comment (tsuyoshicho) [reply README.md]: Thanks suggestion.

NOTE:
Old Base32,Base64 implement RFC 4648's basic mode.
Now add imlement RFC 4648's variant : Base32 Hex,Crockford / Base64 URL safe
Vital.Data.Base32 wrapped Vital.Data.Base32.RFC4648
Vital.Data.Base64 wrapped Vital.Data.Base64.RFC4648

Change:
README.md remove "(RFC Wrapper IF)", only written in vital.txt

**Comment by tsuyoshicho** [comment]
Treat document and changes

**Comment by ujihisa** [comment]
:eyes: > CI

**Comment by tsuyoshicho** [comment]
Sorry for CI error.

Error is two reason.

1. binary literal error in vimlparser -> workaround: change to hex literal.
2. MacVim latest Async.Promise fail error see #698

(first time error, reviewdog result do not see in token error page...)

**Comment by tsuyoshicho** [comment]
Retried and all pass.

**Comment by tsuyoshicho** [comment]
Thanks.

**Pull request merged**